File: Cpp/985_Sum_of_Even_Numbers_After_Queries.cpp
class Solution {
public:
    vector<int> sumEvenAfterQueries(vector<int>& A, vector<vector<int>>& queries) {
        vector<int> result(queries.size());
        int sum_of_even = 0;
        for(auto& num : A){
            sum_of_even += num & 1 ? 0 : num;
        }
        
        for(int i=0; i<queries.size(); ++i){
            int index = queries[i][1];
            int num = A[index] + queries[i][0];
            if(!(num & 1)){
                sum_of_even += num;
            }
            if(!(A[index] & 1)){
                sum_of_even -= A[index];
            }
            A[index] += queries[i][0];
            result[i] += sum_of_even;
        }
        
        return result;
    }
};

File: Kotlin/985_Sum_of_Even_Numbers_After_Queries.kt
class Solution {
    fun sumEvenAfterQueries(nums: IntArray, queries: Array<IntArray>): IntArray {
        val result = IntArray(queries.size)
        var sum = 0
        nums.filter{num -> num and 1 == 0}
            .forEach{num ->
            sum += num
        }
        
        for(i in queries.indices){
            val index = queries[i][1]
            val num = nums[index] + queries[i][0]
            if(num and 1 == 0){
                sum += num
            }
            if(nums[index] and 1 == 0){
                sum -= nums[index]
            }
            nums[index] += queries[i][0]
            result[i] += sum
        }
        
        return result
    }
}